rfcs/0007-filtering.md

```diff
@@ -172,26 +176,36 @@ Fields that represent objects are recursed into, applying the same field selecti
 [Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
 
 How much will this save us?
- 
-Proposed profiling is invoking BFD with at least 10,000 EOB Requests by Patient ID (ExplanationOfBenefit?patient=) and to measure
-the aggregate time of completion of all calls.  In other words, sum up the time taken by each call over all calls.
 
-This test will be baselined on current implementation and then ran again after the solution is deployed.
+As part of a larger effort to reduce job times from days to minutes, AB2D is constructing a performance test harness
```

Comment:
Fantastic!

Reply:
Not sure I really understand the issue(s) here other than to glean from the RFC and various comments.As far as I can tell it's a case of who bears the brunt of data manipulation / filtering when stock content is too verbose for some consumers; as noted, it's why GraphQL has become so popular...it solves this issue nicely!

`AB2D receives more data than needed from BFD when we make API calls to gather EOBs (Explanation of Benefits).  AB2D currently filters out irrelevant fields`

What exactly does this mean?

- Is it the structure of the EOB itself? And AB2D wants an altered EOB structure?
- The EOB structure is fine, AB2D just wants lots (much) of the data nulled out?

_**Question - Who does work reducing fluff?**_

If BFD:
  - hide under feature flag
  - still amounts to BFD compute cycles to reduce payload to minimize fluff; possible solutions:
    1) add filtering mechanism to BFD code; possibles are:
       -  "if" statements in code
       - whitelist (acceptable) map entries
       - affects (gums up?) BFD codebase
    2) use post-processor - minimal (if any) impact on current BFD codebase
       - adds an 'extra' step on way out door
       - extensible
       - no current model / mechanism in place
    3) does this align with long-term goals of BFD?
       - expect other requests like this?

If AB2D:
  - do they have impact assessment numbers on cost of current payload
    - are the requested changes enough to move the needle?
    - if BFD considers a post-processor, did AB2D consider a pre-processor?